src/viam/app/app_client.py

```diff
@@ -0,0 +1,791 @@
+        response: GetRobotPartsResponse = await self._location_client.GetRobotParts(request, metadata=self._metadata)
+        return [RobotPart.from_proto(robot_part=part) for part in response.parts]
+
+    async def get_robot_part(self, robot_part_id: Optional[str], dest: Optional[str] = None, indent: Optional[int] = None) -> RobotPart:
+        """Get a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get.
+            dest (Optional[str]): Optional filepath to write the robot part's config file in JSON format to.
+            indent (Optional[int]): Optional size (in number of spaces) of indent when writing config to `dest`. Defaults to 4.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+
+        Returns:
+            viam.app.app_client.RobotPart: The robot part.
+        """
+        indent = indent if indent else 4
+        request = GetRobotPartRequest(id=robot_part_id)
+        response: GetRobotPartResponse = await self._location_client.GetRobotPart(request, metadata=self._metadata)
+
+        if dest:
+            try:
+                file = open(dest, "w")
+                file.write(f"{json.dumps(json.loads(response.config_json), indent=indent)}")
+            except Exception as e:
+                LOGGER.err(f"Failed to write config JSON to file {dest}", exc_info=e)
+
+        return RobotPart.from_proto(robot_part=response.part)
+
+    async def get_robot_part_logs(
+        self,
+        robot_part_id: str,
+        filter: Optional[str] = None,
+        dest: Optional[str] = None,
+        errors_only: Optional[bool] = None,
+        num_pages: Optional[int] = None,
+    ) -> List[LogEntry]:
+        """Get the logs associated with a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get logs from.
+            filter (Optional[str]): Only include logs with messages that contain the string `filter`. Defaults to empty string "" (i.e., no
+                filter).
+            dest (Optional[str]): Optional filepath to write the log entries to.
+            errors_only (Optional[bool]): Optional boolean specifying whether or not to only include error logs. Defaults to True.
+            num_pages: Optional number of pages of logs to return. Passing 0 returns all pages. Defaults to 1. All pages or the first
+                `num_pages` pages will be returned, whichever comes first.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+            AssertionError: If `num_pages` is less than 0.
+
+        Returns:
+            List[viam.app.app_client.LogEntry]: The list of log entries.
+        """
+        num_pages = num_pages if num_pages is not None else 1
+        assert num_pages >= 0
+        errors_only = errors_only if errors_only else True
+        page_token = ""
+        logs = []
+
+        if num_pages == 0:
+            while True:
```

Comment:
(minor) there's a bit of duplication in this code block. I think we could get rid of the `if num_pages == 0: ... else:` logic and reduce this to a single block by creating a `capture_all_logs = num_pages == 0` variable before the `while_true` and referencing that when deciding whether or not to break. Other approaches could work too! Not worth spending _too_ much time thinking on this, but worth considering.